### DISCUSSION TITLE
Efficient transpose of SparseMatrix

### DISCUSSION THREAD

**Discussion by markusrenoldner** [mfem/mfem#3331] [Q&A] [answered]
Dear members of the mfem community,

What is the standard way to explicitly compute the transpose of a 
`mfem::SparseMatrix`
?

The class has an attribute `*At`, but it is proteced, i.e. it can not be accessed from a `main()` function. Further, the class offers a method called `MultTranspose()` which computes the matrix-vector-product with the transposed matrix. This does not help me, as I am assembling a larger matrix and therefore need explicit access to the transposed matrix.

Thank you for your help in advance.

**Comment by pazner** [comment, marked as answer]
You can use the function `Transpose(const SparseMatrix&)` which will return a new `SparseMatrix*` containing the transpose:
https://github.com/mfem/mfem/blob/b09cc38df9f1da182a5625c92660737ca02e710a/linalg/sparsemat.hpp#L716-L717

**Reply by markusrenoldner** [reply]
Many thanks for the quick reply, this was very helpful!
